libs/openFrameworks/gl/ofShader.cpp

```diff
@@ -447,13 +458,16 @@ bool ofShader::linkProgram() {
 
 		glLinkProgram(program);
 
-		checkProgramLinkStatus(program);
+		linkStatus = checkProgramLinkStatus(program);
```

Comment:
this can break old code since there's checks somewhere else for bLoaded, for example, line 526:

```
//--------------------------------------------------------------
void ofShader::setUniformTexture(const string & name, int textureTarget, GLint textureID, int textureLocation) const{
	if(bLoaded) {
```

and you can do:
```
shader.load(...);
...
if(shader.isLoaded()){
....
}
```

which will be broken. we could keep bLoaded with the same meaning it has now (everything worked) and add individual flags like bShadersCompiled, bLinked, bBinded that you can query to check if every step individually was correct